src/contracts/ChiToken.sol

```diff
@@ -119,102 +119,130 @@ contract ChiToken is IERC20, ERC20WithoutTotalSupply {
         return totalMinted - totalBurned;
     }
 
-    function mint(uint256 value) public {
-        uint256 offset = totalMinted;
+    // Writes the 36 bytes of subcontract creation code to memory at the specified offset
+    function write_bytecode_to_memory(uint256 offset) internal view {
```

Comment:
This is constant for any given ChiToken deployment right? Maybe this can be implemented as an `immutable` so the code just becomes a `CODECOPY` instead of multiple `MSTORE`s? It might not be worth the effort though.